### ISSUE TITLE
isRegisteredUser returns false when WhatsApp hanged or crashed.

### ISSUE THREAD

**Issue by tasaddaq-rana** [pedroslopez/whatsapp-web.js#1740]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I am facing the problem that isRegisteredUser function returns 'false' in case of WhatsApp crashed, hanged, or google chrome window closed due to unexpected problem.

Based on this, I update the status of the message in DB.
So in this case, my all numbers gets marked as Invalid numbers.

Is there any way to make sure that WhatsApp is ready to check number before calling isRegisteredUser function each time to make sure that we are getting accurate results ?

### Expected behavior

isRegisteredUser returns false only if number is not register but not in case of WhatsApp crash etc

### Steps to Reproduce the Bug or Issue

..

### Relevant Code

        if(req.body.number){

          number = phoneNumberFormatter(req.body.number);

          await client.isRegisteredUser(number).then((isRegistered) => {
           isRegisteredNumber = isRegistered;
         });

         if (!isRegisteredNumber) {
            console.log("not registered");
           return;
         }
  //otherwise send message

        }

### Browser Type

Google Chrome

### WhatsApp Account Type

WhatsApp Business

### Does your WhatsApp account have multidevice enabled?

Yes, I am using Multi Device

### Environment

windows 10

**Comment by reyzeal** [comment]
I have this similar problem. This caused by the `getNumberId` function adding `@c.us` suffix to all of group numbers that should end with `@g.us`. So, I did change a single line in `src/Client.js` line 1040 from this:
```js
if (!number.endsWith('@c.us')) {
            number += '@c.us';
        }
```
to this, preventing double suffix like `@g.us@c.us` :
```js
if (!number.endsWith('@g.us') && !number.endsWith('@c.us')) {
            number += '@c.us';
        }
```

**Comment by PurpShell** [comment]
> I have this similar problem. This caused by the `getNumberId` function adding `@c.us` suffix to all of group numbers that should end with `@g.us`. So, I did change a single line in `src/Client.js` line 1040 from this:
> 
> ```js
> if (!number.endsWith('@c.us')) {
>             number += '@c.us';
>         }
> ```
> 
> to this, preventing double suffix like `@g.us@c.us` :
> 
> ```js
> if (!number.endsWith('@g.us') && !number.endsWith('@c.us')) {
>             number += '@c.us';
>         }
> ```

I don't think you should run isRegisteredUser on @g.us, as the group is 100% there if you have an ID for it, don't mod code.

**Comment by PurpShell** [comment]
> ### Is there an existing issue for this?
> 
>     * [x]  I have searched the existing issues
> 
> 
> ### Describe the bug
> 
> I am facing the problem that isRegisteredUser function returns 'false' in case of WhatsApp crashed, hanged, or google chrome window closed due to unexpected problem.
> 
> Based on this, I update the status of the message in DB. So in this case, my all numbers gets marked as Invalid numbers.
> 
> Is there any way to make sure that WhatsApp is ready to check number before calling isRegisteredUser function each time to make sure that we are getting accurate results ?
> ### Expected behavior
> 
> isRegisteredUser returns false only if number is not register but not in case of WhatsApp crash etc
> ### Steps to Reproduce the Bug or Issue
> 
> ..
> ### Relevant Code
> 
> ```
>     if(req.body.number){
> 
>       number = phoneNumberFormatter(req.body.number);
> 
>       await client.isRegisteredUser(number).then((isRegistered) => {
>        isRegisteredNumber = isRegistered;
>      });
> 
>      if (!isRegisteredNumber) {
>         console.log("not registered");
>        return;
>      }
> ```
> 
> //otherwise send message
> 
> ```
>     }
> ```
> 
> ### Browser Type
> 
> Google Chrome
> ### WhatsApp Account Type
> 
> WhatsApp Business
> ### Does your WhatsApp account have multidevice enabled?
> 
> Yes, I am using Multi Device
> ### Environment
> 
> windows 10
> ### Additional context
> 
> _No response_

You shouldn't run any operation while the browser is crashed, restart it using .destroy and .initialize on Client, and then do your action, you can use the .pupBrowser or .pupPage to detect crashes, and the disconnected event. Also you can put all your requests in queue like bee-queue and process them for redundancy

**Comment by PurpShell** [comment]
I don't see this as an issue with the library, this is just like calling .sendMessage while a session hasn't been initialized, you need to make sure the session is OK from your end.
